phlib/kphcomms.c

```diff
@@ -46,74 +46,6 @@ PH_FREE_LIST KphpCommsReplyFreeList;
 #define KPH_COMMS_THREAD_SCALE  2
 #define KPH_COMMS_MAX_MESSAGES  1024
 
-// rev
-typedef struct _FILTER_PORT_EA
-{
-    PUNICODE_STRING PortName;
-    PUNICODE_STRING64 PortName64;
-    USHORT SizeOfContext;
-    BYTE Padding[6]; // not-used (uninitialized heap bytes)
-    BYTE ConnectionContext[ANYSIZE_ARRAY];
-} FILTER_PORT_EA, *PFILTER_PORT_EA;
-
-#define FLT_PORT_CONTEXT_MAX 0xFFE8
-
-// FILE_FULL_EA_INFORMATION (symbols)
-typedef struct _FILTER_PORT_FULL_EA
-{
-    ULONG NextEntryOffset; // 0
-    UCHAR Flags;           // 0
-    UCHAR EaNameLength;    // sizeof(FLT_PORT_EA_NAME) - sizeof(ANSI_NULL)
-    USHORT EaValueLength;  // RTL_SIZEOF_THROUGH_FIELD(FILTER_PORT_EA, Padding) + SizeOfContext
-    CHAR EaName[8];        // FLTPORT\0
-    FILTER_PORT_EA EaValue;
-} FILTER_PORT_FULL_EA, *PFILTER_PORT_FULL_EA;
-
-#define FLT_PORT_EA_NAME "FLTPORT"
-
-#define FILTER_PORT_EA_SIZE \
-    (sizeof(FILE_FULL_EA_INFORMATION) + (sizeof(FLT_PORT_EA_NAME) - sizeof(ANSI_NULL)))
-#define FILTER_PORT_EA_VALUE_SIZE \
-    RTL_SIZEOF_THROUGH_FIELD(FILTER_PORT_EA, Padding)
-//#define FILTER_PORT_EA_VALUE_OFFSET \
-//    (FIELD_OFFSET(FILE_FULL_EA_INFORMATION, EaName) + sizeof(FLT_PORT_EA_NAME))
-//#define FILTER_PORT_EA_VALUE_SIZE \
-//    (FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.ConnectionContext) - FILTER_PORT_EA_VALUE_OFFSET)
-
-#ifdef _WIN64
-C_ASSERT(FILTER_PORT_EA_SIZE == 19); // 0x13
-C_ASSERT(FILTER_PORT_EA_VALUE_SIZE == 24); // 0x18
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.PortName) == 16); // 0x10
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.PortName64) == 24); // 0x18
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.SizeOfContext) == 32); // 0x20
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.ConnectionContext) == 40); // 0x28
-#else
-C_ASSERT(FILTER_PORT_EA_SIZE == 19); // 0x13
-C_ASSERT(FILTER_PORT_EA_VALUE_SIZE == 16); // 0x18
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.PortName) == 16); // 0x10
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.PortName64) == 20); // 0x14
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.SizeOfContext) == 24); // 0x18
-C_ASSERT(FIELD_OFFSET(FILTER_PORT_FULL_EA, EaValue.ConnectionContext) == 32); // 0x20
-#endif
```

Comment:
We lost the asserts when moving them into `ntioapi.h` - but I'm okay with removing them, they were useful during my initial reversing and testing but they're not that useful anymore 👍

Reply:
Yeah, I figured since the type definition for the EA value is known from the symbols, we can omit the extra checks